internal/server/handlers.go

```diff
@@ -0,0 +1,314 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"strconv"
+
+	"github.com/gostuding/goMarket/internal/server/middlewares"
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx/v5/pgconn"
+	"gorm.io/gorm"
+)
+
+type Storage interface {
+	CheckOrdersStorage
+	Registration(context.Context, string, string, string, string) (int, error)
+	Login(context.Context, string, string, string, string) (int, error)
+	AddOrder(context.Context, int, string) (int, error)
+	GetOrders(context.Context, int) ([]byte, error)
+	GetUserBalance(context.Context, int) ([]byte, error)
+	AddWithdraw(context.Context, int, string, float32) (int, error)
+	GetWithdraws(context.Context, int) ([]byte, error)
+}
+
+type LoginPassword struct {
+	Login    string `json:"login"`
+	Password string `json:"password"`
+}
+
+type Withdraw struct {
+	Order string  `json:"order"`
+	Sum   float32 `json:"sum"`
+}
+
+func isValidateLoginPassword(body []byte) (*LoginPassword, error) {
+	var user LoginPassword
+	err := json.Unmarshal(body, &user)
+	if err != nil {
+		return nil, fmt.Errorf("body convert to json error: %w", err)
+	}
+	if user.Login == "" || user.Password == "" {
+		return nil, errors.New("empty values for registration error")
+	}
+	return &user, nil
+}
+
+func checkOrderNumber(order string) error {
+	initPosition := 0
+	if len(order)%2 > 0 {
+		initPosition = 1
+	}
+	summ := 0
+	for i := 0; i < len(order); i++ {
+		value, err := strconv.Atoi(fmt.Sprintf("%c", order[i]))
+		if err != nil {
+			return fmt.Errorf("order error in position %d: %w", i, err)
+		}
+		if initPosition == i {
+			initPosition += 2
+			summ += (2 * value) % 9 //nolint:gomnd // <- algoritm constants
+		} else {
+			summ += value
+		}
+	}
+	if summ%10 == 0 {
+		return nil
+	}
+	return fmt.Errorf("order control summ error. Order: '%s'", order)
+}
+
+// Register ...
+// @Tags Авторизация
+// @Summary Регистрация нового пользователя в микросервисе
+// @Accept json
+// @Param params body LoginPassword true "Логи и пароль пользователя в формате json"
+// @Router /user/register [post]
+// @Success 200 "Успешная регистрация пользователя"
+// @Header 200 {string} Authorization "Токен авторизации"
+// @failure 400 "Ошибка в теле запроса. Тело запроса не соответствует json формату"
+// @failure 409 "Такой логин уже используется другим пользователем"
+// @failure 500 "Внутренняя ошибка сервиса".
+func Register(ctx context.Context, body, key []byte, remoteAddr, ua string,
```

Comment:
Я советую форматировать функции так:

```suggestion
func Register(
    ctx context.Context,
    body,
    key []byte,
    remoteAddr,
    ua string,
```

Reply:
+